app_backend/github_client.py

```diff
@@ -0,0 +1,333 @@
+                    author=commit_item["commit"]["author"]["name"],
+                    date=commit_item["commit"]["author"]["date"][:10],
+                )
+            )
+        return commits
+
+    def _build_file_stats_summary(self, commit_data: Dict[str, Any]) -> str:
+        """Build file statistics summary from GitHub API response."""
+        files = commit_data.get("files", [])
+        files_changed = len(files)
+        total_additions = sum(f.get("additions", 0) for f in files)
+        total_deletions = sum(f.get("deletions", 0) for f in files)
+
+        # Build summary line
+        diff_stats = [f" {files_changed} files changed"]
+        if total_additions > 0:
+            diff_stats.append(f"{total_additions} insertions(+)")
+        if total_deletions > 0:
+            diff_stats.append(f"{total_deletions} deletions(-)")
+
+        # Build individual file stats
+        file_stats = []
+        for file_info in files:
+            filename = file_info.get("filename", "")
+            additions = file_info.get("additions", 0)
+            deletions = file_info.get("deletions", 0)
+            changes = file_info.get("changes", 0)
+
+            file_line = f" {filename}"
+            if changes > 0:
+                file_line += f" | {changes} "
+                # Use proportional scaling instead of arbitrary limits
+                file_line += self._build_change_indicator(additions, deletions)
+            file_stats.append(file_line)
+
+        # Combine file stats with summary
+        result = "\n".join(file_stats)
+        if diff_stats:
+            result += "\n " + ", ".join(diff_stats)
+        return result
+
+    def _build_diff_content(self, commit_data: Dict[str, Any]) -> str:
+        """Build unified diff content from GitHub API response."""
+        diff_parts = []
+        
+        for file_info in commit_data.get("files", []):
+            filename = file_info.get("filename", "")
+            patch = file_info.get("patch", "")
+
+            if patch:
+                # Use more robust diff header construction
+                diff_header = self._build_diff_header(filename, file_info)
+                diff_parts.extend([diff_header, patch, ""])
+
+        return "\n".join(diff_parts)
+
+    def _build_diff_header(self, filename: str, file_info: Dict[str, Any]) -> str:
+        """
+        Build a proper git diff header for a file.
+        
+        Args:
+            filename: The file name
+            file_info: File information from GitHub API
+            
+        Returns:
+            Formatted diff header
+        """
+        # Handle different file statuses (added, deleted, renamed, etc.)
+        status = file_info.get("status", "modified")
+        
+        if status == "added":
+            return f"diff --git a/{filename} b/{filename}\nnew file mode 100644\nindex 0000000..{file_info.get('sha', 'unknown')[:7]}\n--- /dev/null\n+++ b/{filename}"
```

Comment:
How did this string pass formatting? I thought we had maximum length.

Make sure that all this formatting has easily readable unit tests